types/project.go

```diff
@@ -93,30 +93,56 @@ func (p Project) ConfigNames() []string {
 	return names
 }
 
+func (p Project) GetByContainerName(names ...string) (Services, error) {
+	if len(names) == 0 {
+		return p.Services, nil
+	}
+	services := Services{}
+outLoop:
+	for _, name := range names {
+		for _, s := range p.Services {
+			if name == s.ContainerName {
+				services = append(services, s)
+				continue outLoop
+			}
+		}
+		return nil, fmt.Errorf("service with container_name %q could not be found", name)
+	}
+	return services, nil
+}
+
 // GetServices retrieve services by names, or return all services if no name specified
-func (p Project) GetServices(names []string) (Services, error) {
+func (p Project) GetServices(names ...string) (Services, error) {
```

Comment:
Note this refactor to vargs. `GetService` was kept as a comfort function to avoid the collection.